EIPS/eip-2666.md

```diff
@@ -0,0 +1,136 @@
+  - `BNMUL` precompile (address `0x07`) should be repriced from `6000` gas to `6300` gas
+- Keccak256 built-in function was priced as `30` gas plus `6` gas per `32` bytes of the input. Now it should be priced as `13 + (len(input)/136 + 1)*15`
+
+This EIP ideally requires that `MODEXP` repricing is [implemented](https://eips.ethereum.org/EIPS/eip-2565) to also accurately reflect that there is implicit compensation for an old `STATICCALL` cost (pre-2046).
+
+### Benchmarks on the most popular clients
+
+Necessary benchmarks for EIP 2666 were provided by the clients and raw form is assembled in [here](https://docs.google.com/spreadsheets/d/1aCQnk7prrp3Mbcf011BE5zZnkbc3Iw7QAixn6mLbKS0/edit?usp=sharing)
+
+
+Short summary:
+- BNADD
+
+| |   | Gas |
+|---| ---|---|
+Geth |   | 350
+OE |   | 225
+Besu |   | 165
+Nethermind |   | 120
+
+BNADD cost will need to be adjusted when EIP 2046 is accepted
+
+- BNMUL
+
+| |   | Gas |
+|---| ---|---|
+Geth |   | 3000
+OE |   | 6250
+Besu |   | 19109
+Nethermind |   | 5250
+
+Besu result is surprising and is not taken into account. Only little repricing will be necessary when EIP 2046 is accepted
+
+- SHA256 precompile
+
+Currently it's 60 gas + 12 gas per 32 byte word. Proposed formula is `A * ((len(input) + 8) / 64 + 1) + B`, with coefficients below
+
+|   |   | A  | B  |
+|---|---|---|---|
+| Geth  |   | 5  | 3  |
+| OE  |   | 9  | 4  |
+| Besu  |   | 5  | 10  |
+| Nethermind  |   | 10  | 5  |
+
+EIP 2666 proposes `A = 9`, `B = 10`. There are no large one-off costs in this precompile, so it's EIP 2046 - safe.
+
+- RIPEMD precompile
+
+Currently it's 600 gas + 120 gas per 32 byte word. Proposed formula is `A * ((len(input) + 8) / 64 + 1) + B`, with coefficients below
+
+|   |   | A  | B  |
+|---|---|---|---|
+| Geth  |   | 12  | 6  |
+| OE  |   | 8  | 2  |
+| Besu  |   | 29  | 16  |
+| Nethermind  |   | 10  | 6  |
+
+EIP 2666 proposes `A = 12`, `B = 6`. There are no large one-off costs in this precompile, so it's EIP 2046 - safe. Besu expects to have performance improvements by the end of the year.
+
+- Keccak256 performance
+
+Currently it's 30 gas + 6 gas per 32 byte word. Proposed formula is `A * (len(input) / 136 + 1) + B`, with coefficients below
+
+|   |   | A  | B  |
+|---|---|---|---|
+| Geth  |   | 13  | 13  |
+| OE  |   | 15  | 2  |
+| Besu  |   | 19  | 28  |
+| Nethermind  |   | 16  | 3  |
+
+EIP 2666 proposes `A = 15`, `B = 13`. There are no large one-off costs in this precompile, so it's EIP 2046 - safe. Besu expects to have performance improvements by the end of the year.
+
+### Inclusion plan
+
+EIP 2046 can not be accepted without at least part of the repricings from this EIP (namely for `BNADD` and `BNMUL`). Even though work necessary to activate this EIP is very small (adjust few constants), one can also consider two step plan that can be implemented as the following:
+- Include EIP 2046 and `BNADD`/`BNMUL` repricings
+- Include repricings for `SHA256`, `RIPEMD` and `Keccak256`
```

Comment:
```suggestion
As of FORK_BLOCK_NUMBER, `SHA256` precompile (address `0x02`) will costs `10 + ((len(input) + 8)/64 + 1) * 9` gas.
As of FORK_BLOCK_NUMBER, `RIPEMD` precompile (address `0x03`) will cost `6 + ((len(input) + 8)/64 + 1) * 12` gas.
As of FORK_BLOCK_NUMBER, `BNADD` precompile (address `0x06`) will cost `350` gas.
As of FORK_BLOCK_NUMBER, `BNMUL` precompile (address `0x07`) will cost `6300` gas.
As of FORK_BLOCK_NUMBER,  `SHA3` (`0x20`) will cost `13 + (len(input)/136 + 1)*15` gas.
```
Specification section should focus entirely on the specification, and not discuss rationale, motivation, inclusion process, etc.  Someone trying to build a client should be able to read the specification and get exactly what they need without having to dig through a bunch of other information.

A lot of the information included in this section is valuable, but it should be moved to the Rationale section, which describes *why* you chose the values you did.  The specification section just asserts the values.  As far as the historic values, this isn't particularly relevant to the specification itself so I would include them in either the motivation section or just remove that data all together (someone who wants that information can find it if they need to by looking at old specifications or the yellow paper.